code/modules/clothing/masks/breath.dm

```diff
@@ -36,3 +36,31 @@
 	item_state = "m_mask"
 	permeability_coefficient = 0.01
 	put_on_delay = 10
+
+//Vaporizing breath mask
+
+
+/obj/item/clothing/mask/vape/vapormask
+	name = "vaporizing breath mask"
+	desc = "A breath mask with an integrated chemical vaporizer. A label reads \"Warning: Do not fill with flammable materials.\""
+	name = "Vaporizing Breath Mask"
+	desc = "A breath mask with an integrated chemical vaporizer."
+	icon_state = "vapor" 
+	item_state = "vapor"
+	chem_volume = 50 //Lower capacity than a regular vape
+	flags = MASKINTERNALS
+	flags_cover = MASKCOVERSMOUTH
+	gas_transfer_coefficient = 0.10
+	permeability_coefficient = 0.50
+
+/obj/item/clothing/mask/vape/vapormask/Initialize()
+	. = ..() //I have no idea what this line does but apparently Initialize() is supposed to have it.
+	create_reagents(chem_volume)

```

Comment:
create and fill happen in `vape/Initialize()`. At present the mask would start full of nicotine...